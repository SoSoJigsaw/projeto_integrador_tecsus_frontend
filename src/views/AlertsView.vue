<template>
  <transition name="fade" mode="out-in">
    <div v-if="isVisible" class="container-principal">
      <h1>Alertas</h1>
    </div>
  </transition>
</template>

<script lang="ts">
export default {
  data() {
    return {
      isVisible: false
    }
  },
  mounted() {
    setTimeout(() => {
      this.isVisible = true
    }, 1000)
  }
}
</script>

<style>
@import '../assets/base.css';

.container-principal {
  margin-left: 4%;
  margin-top: 3%;
  margin-right: 4%;
  margin-bottom: 3%;
}

.container-principal h1 {
  color: #776ae3;
  font-size: 6vmin;
  font-weight: 400;
}
</style>
