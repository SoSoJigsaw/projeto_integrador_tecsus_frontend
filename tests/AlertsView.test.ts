import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import AlertsView from '@/views/AlertsView.vue'

describe('AlertsView.vue', () => {
  it('should toggle isVisible after 1 second', async () => {
    vi.useFakeTimers()
    const wrapper = mount(AlertsView)
    expect(wrapper.find('.container-principal').exists()).toBe(false)
    expect(wrapper.find('.entre-paginas').exists()).toBe(true)

    // Move on the time to 1000ms
    vi.advanceTimersByTime(1000)

    // Waiting for DOM update
    await wrapper.vm.$nextTick()

    expect(wrapper.find('.container-principal').exists()).toBe(true)
    expect(wrapper.find('.entre-paginas').exists()).toBe(false)

    vi.useRealTimers()
  })
})

it('should mount the component correctly', () => {
  const wrapper = mount(AlertsView)
  expect(wrapper.exists()).toBe(true)
})

it('should start with isVisible set to false', () => {
  const wrapper = mount(AlertsView)
  expect(wrapper.vm.isVisible).toBe(false)
})

it('should display the main container after 1 second', async () => {
  vi.useFakeTimers()
  const wrapper = mount(AlertsView)
  vi.advanceTimersByTime(1000)
  await wrapper.vm.$nextTick()
  expect(wrapper.find('.container-principal').isVisible()).toBe(true)
  vi.useRealTimers()
})

it('should initially show the bell icon', () => {
  const wrapper = mount(AlertsView)
  expect(wrapper.find('.fa-bell').exists()).toBe(true)
})

it('should hide the bell icon after 1 second', async () => {
  vi.useFakeTimers()
  const wrapper = mount(AlertsView)
  vi.advanceTimersByTime(1000)
  await wrapper.vm.$nextTick()
  expect(wrapper.find('.fa-bell').exists()).toBe(false)
  vi.useRealTimers()
})

it('should display the "Alertas" heading after 1 second', async () => {
  vi.useFakeTimers()
  const wrapper = mount(AlertsView)
  vi.advanceTimersByTime(1000)
  await wrapper.vm.$nextTick()
  expect(wrapper.find('.container-principal h1').text()).toContain('Alertas')
  vi.useRealTimers()
})

it('should react to a manual change in isVisible', async () => {
  const wrapper = mount(AlertsView)
  wrapper.vm.isVisible = true
  await wrapper.vm.$nextTick()
  expect(wrapper.find('.container-principal').isVisible()).toBe(true)
})

it('should render without errors with default props', () => {
  const wrapper = mount(AlertsView)
  expect(wrapper.html()).toMatchSnapshot()
})
