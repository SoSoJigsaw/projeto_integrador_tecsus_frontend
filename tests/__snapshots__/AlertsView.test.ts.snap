// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AlertsView.vue > should render without errors with default props 1`] = `
"<main>
  <!--v-if-->
</main>"
`;
